### ISSUE TITLE
ES6 rewrite?

### ISSUE THREAD

**Issue by rwaldron** [meltingice/CamanJS#174]
@meltingice I see you're actively working on this again—what do you think about an ES6 rewrite? 

- Get's away from CS
- Can be transpiled as CS currently is
- Will eventually work without transpilation

**Comment by meltingice** [comment]
Yeah it's definitely something I've been thinking about. Looking ahead, I think it makes sense, and will help to improve community involvement since a lot of people seem to be scared off by Coffeescript's syntax.

**Comment by meltingice** [comment]
I've also been playing with a completely new API for it, but I'm unsure if I want to go down that road yet. Check out the [README](https://github.com/meltingice/CamanJS/tree/cleanup) from the `cleanup` branch.

**Comment by fesor** [comment]
I would like to help with ES6 rewrite. But i need to know, which branch should be considered as feature version? Any roadmap?

What i suggest is divide this work into three steps:

 - To use gulp instead of Cakefile
 - Rewrite tests firstly and use babel for transpiling them
 - Make feature freeze and rewrite library

Then this can be used as a base for future verson.
